src/Stratis.Bitcoin.Features.BlockStore/BlockStoreCache.cs

```diff
@@ -34,18 +37,50 @@ public class BlockStoreCache : IBlockStoreCache
         /// <summary>Provider of time functions.</summary>
         protected readonly IDateTimeProvider dateTimeProvider;
 
+        /// <summary>The blocks count that cache can contain.</summary>
+        public readonly int MaxCacheBlocksCount;
+
+        /// <summary>Entry options for adding blocks to the cache.</summary>
+        private readonly MemoryCacheEntryOptions blockEntryOptions;
+
+        /// <summary>Entry options for adding transactions to the cache.</summary>
+        private readonly MemoryCacheEntryOptions txEntryOptions;
+
         public BlockStoreCache(IBlockRepository blockRepository, IDateTimeProvider dateTimeProvider, ILoggerFactory loggerFactory)
-            : this(blockRepository, new MemoryCache(new MemoryCacheOptions()), dateTimeProvider, loggerFactory)
+            : this(blockRepository, dateTimeProvider, loggerFactory, null)
         {
         }
 
-        public BlockStoreCache(IBlockRepository blockRepository, IMemoryCache memoryCache, IDateTimeProvider dateTimeProvider, ILoggerFactory loggerFactory)
+        public BlockStoreCache(
+            IBlockRepository blockRepository,
+            IDateTimeProvider dateTimeProvider,
+            ILoggerFactory loggerFactory,
+            NodeSettings nodeSettings,
+            IMemoryCache memoryCache = null)
         {
             Guard.NotNull(blockRepository, nameof(blockRepository));
-            Guard.NotNull(memoryCache, nameof(memoryCache));
 
-            this.blockRepository = blockRepository;
+            this.MaxCacheBlocksCount = nodeSettings.ConfigReader.GetOrDefault("maxCacheBlocksCount", 300);
+
             this.cache = memoryCache;
+            if (this.cache == null)
+            {
+                var memoryCacheOptions = new MemoryCacheOptions()
+                {
+                    //We treat one block to be of '100' size and tx to be '1'.
```

Comment:
that makes no sense, you should have these separated if you don't want to use real size

Reply:
We don't really put many tx-blockid links in the cache so it wasn't reasonable to create a separate cache for that so I just specified the lowest size.

Reply:
it feels very arbitrary, so can you explain more on why is this needed, where is it used etc?

Reply:
It is used for 'GetBlockByTrxAsync' that will try to find a txId-blockId in the cache and if not succeeded it will call 'blockRepository.GetTrxBlockIdAsync' and get the block from there. After that block and txId-blockId item will be cached. 

As a matter of fact this is something that we do not use at all but just have implementation for so we can easily get away with deleting this.

Reply:
Fixed.

Reply:
yeah, if it is not used, delete it
it does not make much sense to me to mix tx cache with block cache

Reply:
>it does not make much sense to me to mix tx cache with block cache
Agreed.

>yeah, if it is not used, delete it

Already did (d2ef40e)